pulp_ansible/tests/functional/api/collection/test_sync.py

```diff
@@ -136,3 +150,42 @@ def test_sync_collection_with_stranger_version_numbers_to_check_comparisons(self
 
         content = self.cv_api.list(repository_version=f"{repo.pulp_href}versions/1/")
         self.assertGreaterEqual(len(content.results), 5)
+
+
+class FullDependenciesSync(TestCaseUsingBindings, SyncHelpersMixin):
+    """Collection sync tests for syncing collections and their dependencies."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up class variables."""
+        super().setUpClass()
+        cls.col_api = AnsibleCollectionsApi(cls.client)
+
+    def _clean_database(self):
+        """Cleans the database."""
+        cli_client = cli.Client(cfg)
+        cmd = get_psql_smash_cmd(TRUNCATE_TABLES_QUERY_BASH)
+        cli_client.run(cmd, sudo=True)
+
+    def test_simple_one_level_dependency(self):
```

Comment:
For this I recommend manually, locally creating 3 test collections, A -> B -> C and trying to have the test sync A and asserting A, B, and C are present. I think it'll give more control to use manufactured test data. The test would upload A, B, and C to Pulp, expose via distribution, create a second repo, and then sync A to the second repo (so the sync occurs from pulp). There are other tests that do this so that can provide most of the example code.

Reply:
maybe we can use: https://github.com/peaqe/orion-utils
https://github.com/pulp/pulp_ansible/blob/master/pulp_ansible/tests/functional/api/collection/v3/test_collection.py#L25